cmake/stm32/common.cmake

```diff
@@ -222,6 +222,20 @@ endfunction()
 function(stm32_add_linker_script TARGET VISIBILITY SCRIPT)
     get_filename_component(SCRIPT "${SCRIPT}" ABSOLUTE)
     target_link_options(${TARGET} ${VISIBILITY} -T "${SCRIPT}")
+
+    get_target_property(TARGET_TYPE ${TARGET} TYPE)
+    if(TARGET_TYPE STREQUAL "INTERFACE_LIBRARY")
+        set(INTERFACE_PREFIX "INTERFACE_")
+    endif()
+
+    get_target_property(LINK_DEPENDS ${TARGET} ${INTERFACE_PREFIX}LINK_DEPENDS)
+    if(LINK_DEPENDS STREQUAL "LINK_DEPENDS-NOTFOUND")
+        set(LINK_DEPENDS "${SCRIPT}")
+    else()
+        list(APPEND LINK_DEPENDS "${SCRIPT}")
+    endif()

```

Comment:
```suggestion
    if(LINK_DEPENDS)
        list(APPEND LINK_DEPENDS "${SCRIPT}")        
    else()
        set(LINK_DEPENDS "${SCRIPT}")
    endif()

```

Reply:
The suggested if statement will always evaluate to `true`.

Reply:
Actually it works
https://cmake.org/cmake/help/latest/command/if.html
`False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND`

Reply:
It seems that @Hish15 is right. I didn't know either.
https://cmake.org/cmake/help/latest/command/if.html

Reply:
Interesting... And kind of awkward as well. Suggestion accepted but I feel we are wasting our time since this project is unmaintained. Especially you guys, that review all these PRs that never get merged anyway.

Reply:
We are not exactly wasting our time. Obko has no time to maintain for the moment but since yesterday @Hish15 and me are officially collaborator to the project. We can and will now gradually merge all the PRs. It will take some time though.

Reply:
Glad to hear it.